### ISSUE TITLE
TypeError!

### ISSUE THREAD

**Issue by AIQuantRobot** [numba/numba#8408]
```
import numpy as np
from numba import jit

@jit(nopython=True)
def slidingWindow(a,window):
	'''
	Make an ndarray with a sliding window of the first dimension(axis = 0)
	Parameters
	----------
	a : array_like
		Array to add sliding window to
	window : int
		Size of sliding window

	Returns
	-------
	Array that is a view of the original array with a added dimension of size w.
	'''
	if window < 1:
		raise Exception('Arg: window must be at least 1.')

	if window > a.shape[0]:
		raise Exception('Arg: window cannot bigger than a.shape[0]')

	shape = (a.shape[0] - window + 1,) + a.shape[1:] + (window,)
	strides = a.strides + (a.strides[0],)

	return np.lib.stride_tricks.as_strided(a, shape=shape, strides=strides)

M1 = np.random.randint(1000,size=(300,3000))
M1_3D_21 = slidingWindow(M1,21)
n1,n2,n3 = M1_3D_21 .shape[0],M1_3D_21 .shape[1],M1_3D_21 .shape[2]
M1_3D_21_argsort  =  np.argsort(M1_3D_21 ,axis=2)   #jit no support for 'axis'

@jit(nopython = True)
def PIMean_mid(m_3d,argsortTensor,n1,n2):
	'''

	'''
	PIMeanMat = np.full(fill_value=np.nan,shape=(n1,n2))
	for i in range(n1):
		for j in range(n2):
			sorted_ij = m_3d[i,j][argsortTensor[i,j]]
			PIMeanMat[i,j] = np.mean(sorted_ij[p1idx:(p2idx+1)])

	return PIMeanMat

ans = PIMean_mid(M1_3D_21 ,M1_3D_21_argsort  ,n1,n2)
```

```
Traceback (most recent call last):

  File "C:\Users\isaac\AppData\Local\Temp/ipykernel_13308/3720963792.py", line 1, in <module>
    PIMean_mid(M1_3D_21,argsortTensor1,M1_3D_21.shape[0],M1_3D_21.shape[1])

  File "D:\ProgramFiles\Anaconda3\lib\site-packages\numba\core\dispatcher.py", line 482, in _compile_for_args
    error_rewrite(e, 'typing')

  File "D:\ProgramFiles\Anaconda3\lib\site-packages\numba\core\dispatcher.py", line 423, in error_rewrite
    raise e.with_traceback(None)

TypingError: No implementation of function Function(<function full at 0x000001B0083E5700>) found for signature:
 
full(fill_value=float64, shape=UniTuple(int64 x 2))
 
There are 4 candidate implementations:
  - Of which 4 did not match due to:
  Overload in function '_OverloadWrapper._build.<locals>.ol_generated': File: numba\core\overload_glue.py: Line 131.
    With argument(s): '(fill_value=float64, shape=UniTuple(int64 x 2))':
   Rejected as the implementation raised a specific error:
     LoweringError: Failed in nopython mode pipeline (step: native lowering)
   Cannot cast float64 to UniTuple(int64 x 2): double %"arg.fill_value"
   
   File "<string>", line 3:
   <source missing, REPL/exec in use?>
   
   During: lowering "$8call_function.3 = call $2load_global.0(fill_value, shape, func=$2load_global.0, args=[Var(fill_value, <string>:3), Var(shape, <string>:3)], kws=(), vararg=None, target=None)" at <string> (3)
  raised from D:\ProgramFiles\Anaconda3\lib\site-packages\numba\core\errors.py:786

During: resolving callee type: Function(<function full at 0x000001B0083E5700>)
During: typing of call at C:\Users\isaac\AppData\Local\Temp/ipykernel_13308/2890536856.py (6)
```

**Comment by gmarkall** [comment]
For me the reproducer gives:

```
numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
NameError: name 'p1idx' is not defined
```

Which highlights the issue with the reproducer. What version of Numba are you using?

**Comment by stuartarchibald** [comment]
I suspect it's this bug #7982, the args are presented in a different order to their API definition order i.e. defined `np.full(shape, fill_value)`, presented `np.full(fill_value=<something>, shape=<something>)`. This bug was present in 0.55, but fixed in 0.56 via patch #7996.

**Comment by stuartarchibald** [comment]
Closing this issue due to lack of information. If more information becomes available please add it and reopen the issue. Thanks.
